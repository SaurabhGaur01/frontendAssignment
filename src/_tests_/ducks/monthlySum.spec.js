import {
    setMonthlySum,
    monthlySumReducer,
} from '../../ducks/monthlySum';

const mockKey = 'mockValue';

describe('monthlySum Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setMonthlySum should match the snapshot' , () => {
            expect(setMonthlySum(mockKey)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(monthlySumReducer()).toEqual(200);
        });
        it('should return the passed state if called with no action' , () => {
            expect(monthlySumReducer(mockKey)).toEqual(mockKey);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(monthlySumReducer(mockKey, unknownAction)).toEqual(mockKey);
        });
        it('should return the value set' , () => {
            expect(monthlySumReducer('', setMonthlySum(mockKey))).toEqual(mockKey);
        });
    });
});