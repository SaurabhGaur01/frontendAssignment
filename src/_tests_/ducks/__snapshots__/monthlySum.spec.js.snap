// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`monthlySum Duck Tests Action Generator setMonthlySum should match the snapshot 1`] = `
Object {
  "type": "modules/app/SET_MONTHLY_SUM",
  "value": "mockValue",
}
`;
